Example-Energy-REXX/RXENRGY1.rexx

```diff
@@ -0,0 +1,2172 @@
+            ('ERROR: found LOCAL CPC but it did not contain a name property')
+          end
+      end
+ end /* endloop CPCs */
+
+if localCPCfound then
+  do
+    Say
+    Say 'Successfully obtained LOCAL CPC Info:'
+    Say '  name:'||CPCname
+    Say '  uri:'||CPCuri
+    Say '  target-name:'||CPCtargetName
+    Say
+    return 0
+  end
+
+
+/* if we're still here than something went wrong */
+return fatalError('ERROR - ** failed to obtain LOCAL CPC info **')
+
+/*******************************************************/
+/* Function:  GetCPCInfo                               */
+/*                                                     */
+/* Retrieve the uri and target name associated with    */
+/* CPCname and prime CPCuri and CPCtargetName          */
+/* variables with the info                             */
+/*                                                     */
+/* Return 0 if successful, otherwise return -1         */
+/*******************************************************/
+GetCPCInfo:
+
+methodSuccess = TRUE /* assume success */
+emptyCPCResponse = '{"cpcs":[]}'
+
+/* List information for specific CPC
+
+   GET /api/cpcs?name=CPCname
+
+   v1: HWICONN(HWI_CPC,IBM390PS.CPC)
+
+*/
+reqUri = '/api/cpcs?name='||CPCname
+CPCInfoResponse = GetRequest(reqUri)
+
+emptyCPCArray = INDEX(CPCInfoResponse, emptyCPCResponse)
+if emptyCPCArray > 0 | CPCInfoResponse = '' Then
+  do
+    return fatalError('** failed to retrieve CPC **')
+  end
+
+/* Parse the response to obtain the uri
+   and target name associated with CPC
+*/
+call JSON_parseJson CPCInfoResponse
+
+CPCuri = JSON_findValue(0,"object-uri", HWTJ_STRING_TYPE)
+if foundJSONValue(CPCuri) = FALSE then
+  methodSuccess = FALSE
+
+CPCtargetName = JSON_findValue(0,"target-name", HWTJ_STRING_TYPE)
+if foundJSONValue(CPCtargetName) = FALSE then
+  methodSuccess = FALSE
+
+if methodSuccess then
+  do
+    Say
+    Say 'Successfully obtained CPC Info:'
+    Say '  uri:'||CPCuri
+    Say '  target-name:'||CPCtargetName
+    Say
+    return 0
+  end
+
+Say
+Say 'Obtained some or none of the CPC Info:'
+Say '  uri:('||CPCuri||')'
+Say '  target-name:('||CPCtargetName||')'
+Say 'full response body:('||CPCInfoResponse||')'
+Say
+return fatalError(' ** failed to obtain CPC info **')
+
+/*******************************************************/
+/* Function:  QueryEnergy                              */
+/*                                                     */
+/* Retrieve attributes identified by                   */
+/* EnergyAttribute stem variable                       */
+/*                                                     */
+/* Return 0 if successful, otherwise return -1         */
+/*******************************************************/
+QueryEnergy:
+/* Use retrieved CPC uri to retrieve
+   specific attribute properties
+   GET /api/cpcs/{cpc-id}/energy-management-data
+*/
+
+if EnergyAttribute.0 < 1 then
+  do
+    return fatalError('** no attributes to query **')
```

Comment:
maybe change it to no attributes to retrieve? given you're technically not using the '? filter on the uri to query'